### ISSUE TITLE
webpack4 htmlWebpackPluginBeforeHtmlGeneration is undefined

### ISSUE THREAD

**Issue by wZi** [van-nguyen/webpack-cdn-plugin#19]
```
Happy[js]: Version: 5.0.0-beta.3. Threads: 5 (shared pool)
(node:59770) DeprecationWarning: Tapable.plugin is deprecated. Use new API on `.hooks` instead
Happy[js]: All set; signaling webpack to proceed.
~/node_modules/webpack-cdn-plugin/main.js:48
      compilation.hooks.htmlWebpackPluginBeforeHtmlGeneration.tapAsync('WebpackCdnPlugin', (data, callback) => {
TypeError: Cannot read property 'tapAsync' of undefined
```

any solution is helpfully.
Thanks

**Comment by van-nguyen** [comment]
Hey @wZi which version of webpack are you using?

**Comment by wZi** [comment]
Thanks for your replay.
The webpack version is V4.1.1

**Comment by van-nguyen** [comment]
I still can't reproduce this problem, are you able to share you `package.json`?

**Comment by corcoran** [comment]
I ran into this issue while using vue-cli and was able to fix it by explicitly importing HtmlWebpackPlugin and registering it as a plugin.

vue.config.js example:
```
const HtmlWebpackPlugin = require('html-webpack-plugin')
const WebpackCdnPlugin = require('webpack-cdn-plugin')
module.exports = {
  configureWebpack: {
    plugins: [
      new HtmlWebpackPlugin(),
      new WebpackCdnPlugin({
        modules: {
          'vue': [
            {name: 'vue', var: 'Vue', path: 'dist/vue.runtime.min.js'},
          ],
        }
      })
    ]
  },
```

**Comment by van-nguyen** [comment]
There is a PR to support v3 on a legacy branch, this should resolve once merged #37

**Comment by gauravkhuranacopart** [comment]
Well i have observed if we define new WebpackCdnPlugin below new HtmlPlugin the error can be averted
